BD-3069 Diataxis templates (#7367)
braze-inc/braze-docs: _docs/_contributing/page_types.md

---
nav_title: Page Types
article: Page Types
page_order: 4
toc_headers: h2
---

# Page types

> Braze documentation should follow the Diátaxis framework, which organizes documentation into four categories, or types, that each corresponds with a distinct need or learning preference. Each Braze article can contain multiple documentation types, though each type should reside in its own section.

Reviewer: ```suggestion
> Braze Docs follows the [Diátaxis framework](https://diataxis.fr/), which organizes pages into one of four content types, each one meeting a different learning objective. While a single page on Braze Docs may contain multiple content types, each type should get a dedicated section on the page.

These are the four content types you'll find on Braze Docs:
```

Reviewer: *Note: We should probably update the file name / page name to **Content Types** instead of **Page Types** since a single page can have multiple types. Thoughts?

| Documentation type | Purpose |
| --- | --- |
| [How-to guides](#how-to-guides) | Help the user **apply knowledge**. |
| [Tutorials](#tutorials) | Help the user **acquire knowledge**. |
| [References](#references) | Provide the user with **technical knowledge**. |
| [Explanations](#explanations) | Broaden the user’s **contextual knowledge**. |

## How-to guides

> How-to guides are action-based and chronological steps. They focus on telling a user how to achieve a specific and successful result.

Reviewer: ```suggestion
How-to guides are action-based, chronological steps that show users how to complete a specific task.
```

Reviewer: ^ Thought here is just making this one sentence instead of two sentences. You don't have to use my exact edit, but an option for trimming it down.

Examples include:
- [Creating a Content Card]({{site.baseurl}}/user_guide/message_building_by_channel/content_cards/create/)
- [WhatsApp setup]({{site.baseurl}}/user_guide/message_building_by_channel/whatsapp/overview/)
- [Connected sources integration]({{site.baseurl}}/user_guide/data_and_analytics/cloud_ingestion/connected_sources/)

> ### Content guidelines

Reviewer: ```suggestion
### Guidelines
```

> Cover only what the user needs to know to take action. Only include [references](#references) (technical, non-action based information) that are vital to reaching their goal, such as a list of the options a user can select from during a step. Link out to references that are longer than reasonable to include in the same article, such as [Segmentation filters]({{site.baseurl}}/user_guide/engagement_tools/segments/segmentation_filters/).

Reviewer: Thoughts on converting this to a list that can be skimmed? I can help if needed. 👍🏽

Author: Good suggestion! I'll get that added.

The how-to guide should cover the "happy path" to the goal, which assumes the best case scenario while completing the task. Avoid providing troubleshooting information, which you can include in a separate reference section or article.

#### Headers

> Use Heading 2 and action-oriented language for the primary how-to name, such as "Creating a page".

Reviewer: ```suggestion
H2 headings (`##`) should be action-oriented and reflect the general goal for this step.
```

> 

Reviewer: ``````suggestion

```
## Creating a page

Open the relevant directory, then create a new Markdown file for your page.
```

``````

> For how-tos with long or complicated steps, use nested headers to group related steps like so:

Reviewer: ```suggestion
For how-tos with long or complicated steps, use nested headers to group related steps as shown in the following Markdown sample. For short steps, see [Ordered lists](#ordered-lists) instead.
```

{% raw %}
```markdown

> [H2] Creating a page

Reviewer: Let's replace these with the `##` markdown syntax is possible.

[H3] Step 1: Create a new file

Open the relevant directory, then create a new Markdown file for your page.

PAGE_TITLE.md

Replace `PAGE_TITLE` with the title of your page, which should follow the Braze Docs Style Guide. Use all lowercase characters, remove special characters, and replace spaces with underscores (`_`). Your filename should be similar to the following:

- **Page title:** Setting up your development environment for C++
- **File name:** `setting_up_your_development_environment_for_cpp.md`

[H3] Step 2: Add a template

Copy and paste one of the following templates into your Markdown file. For more information, see [Templates]({{site.baseurl}}/contributing/templates/).

[H4] Basic template
```
{% endraw %}

For optional steps, put “(optional)” at the end of a header, such as:

{% raw %}
```markdown
Step 4: Add additional content (optional)
```
{% endraw %}

> #### Bullet points

Reviewer: ```suggestion
#### Ordered lists
```

> For how-tos with short and straightfoward steps, use numbered bullets below headers like so:

Reviewer: ```suggestion
For how-tos with short and straightforward steps, use an ordered list in Markdown format.
```
